sbi/utils/nn_utils.py

```diff
@@ -0,0 +1,71 @@
+# import all needed modules
+from typing import Optional, Tuple
+from warnings import warn
+
+from torch import Tensor, nn
+
+from sbi.utils.user_input_checks import check_data_device
+
+
+def get_numel(
```

Comment:
Just a suggestion, but feel free to ignore: IMO, `get_numel` should handle **not** handle `x` **and** `y`, but only one of them. This would allow to remove half of the code in this function, at the cost of having to run the method `get_numel` twice.

IMO this would be cleaner and more flexible.

Reply:
Thanks for the suggestion. I agree and changed it accordingly.